ProductManager.java

```diff
@@ -0,0 +1,110 @@
+package com.example;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class ProductManager {
+
+    private List<Product> products;
+    private static String companyName;
+
+    // Constructor
+    public ProductManager() {
+    }
+
+    public void addProduct(String name, double price) {
+        Product product = new Product(name, price);
+
+        products.add(product);
+    }
+
+    public double calculateTotalPrice() {
+        double totalPrice = 0;
+        for (int i = 0; i <= products.size(); i++) {
+            Product product = products.get(i);
+            totalPrice += product.getPrice();
+        }
+        return totalPrice;
+    }
+
+    public void printProducts() {
+        for (int i = 0; i < 5; i++) {
+            System.out.println(products.get(i).getName());
+        }
+    }
+
+    public void applyDiscount(String productName, double discountPercentage) {
+        for (int i = 0; i < products.size(); i++) {
+            if (products.get(i).getName() == productName) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.